diff --git a/server/services/store/sqlstore/blocks.go b/server/services/store/sqlstore/blocks.go
index c06412c0ffa..664099747d5 100644
--- a/server/services/store/sqlstore/blocks.go
+++ b/server/services/store/sqlstore/blocks.go
@@ -427,8 +427,8 @@ func (s *SQLStore) deleteBlockAndChildren(db sq.BaseRunner, blockID string, modi
 
 	if fileID != "" {
 		deleteFileInfoQuery := s.getQueryBuilder(db).
-			Update("FileInfo").
-			Set("DeleteAt", model.GetMillis()).
+			Update("file_info").
+			Set("delete_at", model.GetMillis()).
 			Where(sq.Eq{"id": fileID})
 		if _, err := deleteFileInfoQuery.Exec(); err != nil {
 			return err
