diff --git a/webapp/src/mutator.ts b/webapp/src/mutator.ts
index 3dea2ae4103..c616b69ccee 100644
--- a/webapp/src/mutator.ts
+++ b/webapp/src/mutator.ts
@@ -224,8 +224,8 @@ class Mutator {
                 await afterRedo?.(board)
             },
             async () => {
-                await beforeUndo?.(board)
                 await octoClient.undeleteBoard(board.id)
+                await beforeUndo?.(board)
             },
             description,
             this.undoGroupId,
